test/SIL.XForge.Identity.Tests/Controllers/Account/AccountControllerTests.cs

```diff
@@ -97,6 +97,43 @@ public async Task ForgotPassword_IncorrectUsernameOrEmail()
             Assert.IsTrue(model.EnableErrorMessage == true);
         }
 
+        [Test]
+        public async Task ForgotPassword_UpdateResetPasswordKeyAndExpirationDate()
+        {
+            var env = new TestEnvironment();
+            var model = new ForgotPasswordViewModel
+            {
+                UsernameOrEmail = "user",
+                EnableErrorMessage = false
+            };
+            IActionResult result = await env.Controller.ForgotPassword(model);
+            UserEntity user = await env.users.Query().SingleOrDefaultAsync(x => x.Username == model.UsernameOrEmail);
+            Assert.True(user.ResetPasswordKey != "jGc6Qe4i1kgM+aA4LVczTJwfHx2YuDR/", "ResetPasswordKey not saved.");
+            Assert.True(user.ResetPasswordExpirationDate != DateTime.Now, "ResetPasswordExpirationDate not saved.");
+        }
+
+        [Test]
+        public async Task ForgotPassword_SendEmail()
```

Comment:
You don't need to test the `EmailService` class. In the unit test `ForgotPassword_CorrectUsernameOrEmail`, you can check that the `SendEmail` method of the `IEmailService` mock was called with the correct parameters. See the NSubstitute [docs](http://nsubstitute.github.io/help/received-calls/) to see how to check for a received call with specified arguments. I will clarify what I want in the standup meeting.

Reply:
We removed unit test for EmaiService and added EmailServer mock test in CorrectusernameOrEmail unit test. Thank you.